### ISSUE TITLE
Scaling, cropping, of Dodo print service

### ISSUE THREAD

**Issue by masinter** [devhawala/dodo#16]
I know that a full-featured Interpress -> Postscript transition may be out of scope, but just for simple raster image printing from Medley Interlisp: I printed 3 files, two of bitmaps (from a window and a region) and one of text. The text printed beautifully but the images didn't:

https://github.com/Interlisp/medley/issues/931#issuecomment-1283315720

[ipimages.zip](https://github.com/devhawala/dodo/files/9821772/ipimages.zip)
attached is a gzip'd tar labeled as .zip with similar images but written with Interlisp's Postscript implementation.

**Comment by devhawala** [comment]
yes, mapping of bitmap transformation matrices from Interpress to PostScript is a bit difficult for Interlisp produced Interpress: the Interpress interpreter used by Dodo (IPPROC.PS) stems from the GlobalView 2.1 product for Windows 3.x, so it surely is not as generic as one would hope, but more biased towards the Globalview (or maybe even Star?) style of issuing Interpress streams.

There are already 2 or 3 hacks for bitmaps when mapping the Interpress tokens (reordering and modifications) when disassembling Interpress for later PS processing, i hope to find the next hacks to also handle these specific cases without breaking the others...

**Comment by devhawala** [comment]
ok, one of the hacks for correcting the placement of bitmaps in case of Interpress produced by Interlisp/Medley was suboptimal and i found an improvement, the 2 attached PDFs were created with the corrected version of Dodo printing. This corrects only the misplacement of the bitmap stripes issued for large bitmaps, however the scaling factor and rotation for the complete image are as Interlisp/Medley wrote them to the Interpress master.

[job_0184_05BE_AEA5.pdf](https://github.com/devhawala/dodo/files/9847103/job_0184_05BE_AEA5.pdf)
[job_0184_05BE_AEA4.pdf](https://github.com/devhawala/dodo/files/9847104/job_0184_05BE_AEA4.pdf)

That was easier than the fixes for Dodo mailing ([Issue #15](https://github.com/devhawala/dodo/issues/15)), which i'm still working on...

Both sets of fixes will be uploaded together as soon as the mailing fixes have reached some stability.

**Comment by masinter** [comment]
thanks, we should try with a couple of documents that had custom fonts -- the equation editor and a square-dance caller notation. I'll send some .IPs (cc: @interlisp)

**Comment by devhawala** [comment]
i just uploaded a new Dodo version fixing this issue; as always simple download the distribution file `dist.zip` and replace the file `dodoserver-and-nethub.jar` in your Dodo-setup with the one from the distribution file.
